src/ngx_pagespeed.cc

```diff
@@ -544,7 +544,8 @@ ngx_int_t copy_response_headers_to_ngx(
     }
 
     if (STR_EQ_LITERAL(name, "Cache-Control")) {
-      ps_set_cache_control(r, const_cast<char*>(value_gs.c_str()));
+      ps_set_cache_control(r, string_piece_to_pool_string(
+          r->pool, value_gs.c_str()));
```

Comment:
@jeffkaufman Good ideas. I thought about this some ,what do you think about this?: We currently know when an NgxBaseFetch instance has detached from the nginx side (in which case we should not process further events received from psol for that NgxBaseFetch instance).
As such it should be ok to safely defer destruction of NgxBaseFetches. We could add an extra reference for this and decrement one more time from a separate request cleanup handler. If we do that, we don't have to copy strings, and we don't have unsafe values.